Realm.Shared/linq/RealmResultsVisitor.cs

```diff
@@ -254,20 +261,13 @@ protected void VisitCombination(BinaryExpression b,  Action<QueryHandle> combine
 
         internal static object ExtractConstantValue(Expression expr)
         {
-            var constant = expr as ConstantExpression;
-            if (constant != null)
-            {
-                return constant.Value;
-            }
+            var objectMember = Expression.Convert(expr, typeof(object));
```

Comment:
Instead of replacing this completely, could we use `LambdaExpression.Compile` as a third option if the first two conditions fail, just so we don't have to rely on it when the faster options are available?